src/main/resources/com/google/api/codegen/ruby/main.snip

```diff
@@ -280,11 +280,18 @@
 
 @private createResourceFunction(function)
   @# Returns a fully-qualified {@function.entityName} resource name string.
+  @if function.isResourceNameDeprecated
+    @# @@deprecated Multi-pattern resource names will have unified creation and parsing helper functions.
+    @# This helper function will be deleted in the next major version.
+  @end
   @join param : function.resourceIdParams
     @# @@param {@param.docName} [String]
   @end
   @# @@return [String]
   def self.{@function.name} {@createResourceFunctionParams(function.resourceIdParams)}
+    @if function.isResourceNameDeprecated
+      warn "This helper function will be deleted in the next major version."
```

Comment:
Please omit this. Ruby devs do not like warnings printed to stderr with no way of disabling it.

Reply:
Done.